server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckBaseUtility.java

```diff
@@ -32,391 +33,535 @@
+                        ctx.children.get(index),
+                        "Accessory options not allowed when browsing with START or END");
+            } else if (isCoreRule
+                    && index + 1 < ctx.children.size()
+                    && ctx.children.get(index + 1).getText().contains("(")) {
+                // Skip ahead for core rules that require parameters
+                index++;
             }
-          }
-        });
-  }
+        }
+    }
+
+    /**
+     * Ensures the main option has it's required parameter when not browsing with START or END
+     *
+     * @param ctx           Context to evaluate
+     * @param coreRuleIndex The CICSParser index for the core rules with required parameter
+     */
+    public void checkStatementHasParameter(ParserRuleContext ctx, int... coreRuleIndex) {
+        // Make sure the main option has it's required parameter if not browsing with START or END
+        int traversalIndex = 0;
+        for (ParseTree rule : ctx.children) {
+            if (TerminalNode.class.isAssignableFrom(rule.getClass())) {
+                if (ArrayUtils.contains(coreRuleIndex, ((TerminalNode) rule).getSymbol().getType())) {
+                    if (traversalIndex + 1 < ctx.children.size()) {
+                        if (!ctx.children
+                                .get(traversalIndex + 1)
+                                .getText()
+                                .contains("(")) { // TODO: Find a better strategy for detecting parameters
+                            throwBrowsingViolation(
+                                    ctx.children.get(traversalIndex), "Missing required option parameter");
+                        }
+                    } else
+                        throwBrowsingViolation(
+                                ctx.children.get(traversalIndex), "Missing required option parameter");
+                    break;
+                }
+            }
+            traversalIndex++;
+        }
+    }
+
+    /**
+     * Ensures the main option does not have parameter if browsing with START or END
+     *
+     * @param ctx           Context to evaluate
+     * @param coreRuleIndex The CICSParser index for the core rules without parameters
+     */
+    public void checkBrowsingHasNotParameter(ParserRuleContext ctx, int... coreRuleIndex) {
+        // Make sure the main option does not have parameter if browsing with START or END
+        int traversalIndex = 0;
+        for (ParseTree rule : ctx.children) {
+            if (TerminalNode.class.isAssignableFrom(rule.getClass())) {
+                if (ArrayUtils.contains(coreRuleIndex, ((TerminalNode) rule).getSymbol().getType())) {
+                    if (traversalIndex + 1 < ctx.children.size()
+                            && ctx.children
+                            .get(traversalIndex + 1)
+                            .getText()
+                            .contains("(")) { // TODO: Find a better strategy for detecting parameters
```

Comment:
Ideally, the strategy should not require turning the whole context into a string, but just check for CICSLexer.LPARENCHAR token.